user-service/src/main/java/rs/edu/raf/banka1/controller/AuthenticationController.java

```diff
@@ -0,0 +1,56 @@
+package rs.edu.raf.banka1.controller;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.web.bind.annotation.*;
+import rs.edu.raf.banka1.model.Permission;
+import rs.edu.raf.banka1.model.User;
+import rs.edu.raf.banka1.requests.LoginRequest;
+import rs.edu.raf.banka1.responses.LoginResponse;
+import rs.edu.raf.banka1.responses.UserResponse;
+import rs.edu.raf.banka1.services.UserService;
+import rs.edu.raf.banka1.utils.JwtUtil;
+
+import java.util.stream.Collectors;
+@RestController
+@RequestMapping("/auth")
+public class AuthenticationController {
+    private final AuthenticationManager authenticationManager;
+    private final UserService userService;
+    private final JwtUtil jwtUtil;
+
+    public AuthenticationController(AuthenticationManager authenticationManager, UserService userService, JwtUtil jwtUtil) {
+        this.authenticationManager = authenticationManager;
+        this.userService = userService;
+        this.jwtUtil = jwtUtil;
+    }
+
+    @PostMapping("/login")
+    public ResponseEntity<?> login(@RequestBody LoginRequest loginRequest){
+        try {
+            authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(loginRequest.getEmail(), loginRequest.getPassword()));
+        } catch (Exception   e){
+            return ResponseEntity.status(401).build();
+        }
+
+        UserResponse user = this.userService.findByEmail(loginRequest.getEmail());
+        StringBuilder permissions = new StringBuilder();
+
+        //packs permissions into json list
+        // [perm1, perm2...]
+        if (user.getPermissions() != null && !user.getPermissions().isEmpty()) {
```

Comment:
Da li je ovaj format potreban zbog jwt-a, i da li ce se jos negde koristiti? Ako da, bilo bi dobro ekstraktovati ovo u neku util klasu. Mada svakako bih to odmah uradio da ne zagusujemo kontroler.

Reply:
Resolved :)